File: lib/ClientUdp/command/get_room_users_handler.dart
import '../client_state.dart';
import '../constants.dart';
import '../handshake_manager.dart';
import '../json_helper.dart';
import 'command_handler.dart';

class GetRoomUsersHandler implements CommandHandler {
  @override
  void handle(
      String args, ClientState clientState, HandshakeManager handshakeManager) {
    String roomId = args.trim();

    if (clientState.sessionKey == null) {
      print("You must be logged in to get room users. Use /login <id> <pw>");
      // stdout.write("> ");
      return;
    }

    if (roomId.isEmpty) {
      print("Room ID cannot be empty.");
      // stdout.write("> ");
      return;
    }

    Map<String, dynamic> data = {
      Constants.KEY_CHAT_ID: clientState.currentChatId,
      Constants.KEY_ROOM_ID: roomId
    };

    Map<String, dynamic> request =
        JsonHelper.createRequest(Constants.ACTION_GET_ROOM_USERS, data);
    handshakeManager.sendClientRequestWithAck(
        request, Constants.ACTION_GET_ROOM_USERS, clientState.sessionKey!);
  }

  @override
  String getDescription() => Constants.CMD_GET_ROOM_USERS_DESC;
}

File: UdpClientFactor/src/main/java/UdpChatClient/command/GetRoomUsersHandler.java
package UdpChatClient.command;

import com.google.gson.JsonObject;

import UdpChatClient.ClientState;
import UdpChatClient.Constants;
import UdpChatClient.HandshakeManager;
import UdpChatClient.JsonHelper;

public class GetRoomUsersHandler implements CommandHandler {

    @Override
    public void handle(String args, ClientState clientState, HandshakeManager handshakeManager) {
        String roomId = args.trim();

        if (clientState.getSessionKey() == null) {
            System.out.println("You must be logged in to get room users. Use /login <id> <pw>");
            System.out.print("> ");
            return;
        }

        // Validate input
        if (roomId.isEmpty()) {
            System.out.println("Room ID cannot be empty.");
            System.out.print("> ");
            return;
        }

        JsonObject data = new JsonObject();
        data.addProperty(Constants.KEY_CHAT_ID, clientState.getCurrentChatId());
        data.addProperty(Constants.KEY_ROOM_ID, roomId);

        JsonObject request = JsonHelper.createRequest(Constants.ACTION_GET_ROOM_USERS, data);
        handshakeManager.sendClientRequestWithAck(request, Constants.ACTION_GET_ROOM_USERS, clientState.getSessionKey());
        // No need to print "> " here - done in the handshakeManager
    }

    @Override
    public String getDescription() {
        return Constants.CMD_GET_ROOM_USERS_DESC;
    }
}
